database/migrations/2023_12_19_211314_user_unique_email.php

```diff
@@ -4,12 +4,22 @@
 use Illuminate\Database\Schema\Blueprint;
 use Illuminate\Support\Facades\Schema;
 
+use App\Models\User;
+
 return new class extends Migration {
     /**
      * Run the migrations.
      */
     public function up(): void
     {
+        // Delete any duplicate email addresses, keeping the oldest userid for each one.
+        $userids_to_keep = User::select(DB::raw('MIN(id)'))
+             ->groupBy('email')
+             ->pluck('MIN(id)');
+        $num_deleted = User::whereNotIn('id', $userids_to_keep)->delete();
```

Comment:
This will query almost all the users, and then send them back to the DB.  It would be better to combine them using nested subqueries.  For example (note: untested):
```
$num_deleted = User::whereNotIn('id', function ($query) {
    $ query->min('id')
        ->from(User::class)
        ->groupBy('email');
})->delete();
```

There's probably a more efficient `HAVING ...` approach, but I'm not as confident about how that would work with Laravel.